odoo/addons/base/models/ir_ui_view.py

```diff
@@ -2830,7 +2830,7 @@ def _get_records_action(self, **kwargs):
         if len(self) == 0:
             length_dependent = {'views': [(False, 'form')]}
         elif len(self) == 1:
-            length_dependent = {'views': [(False, 'form')], 'res_id': self.id}
+            length_dependent = {'views': [(False, 'form')], 'res_id': self._origin.id}
```

Comment:
It is a nice a idea to improve the robustness of this method. But the method doesn't respect the length of the real ids.
Here a suggestion to reinforce the method when there are new records in self:
```Python
ids = self.ids  # ids actually return the origin ids and filter out ids without origin.
if len(ids) == 0:
    length_dependent = {'views': [(False, 'form')]}
elif len(ids) == 1:
    length_dependent = {'views': [(False, 'form')], 'res_id': ids[0]}
else:
    length_dependent = {
         'views': [(False, 'list'), (False, 'form')],
          'domain': [('id', 'in', ids)]
      }
```